### PR TITLE
Add page for websocket

### LINKED ISSUE #536: Describe how to stream data with WS
- [ ] Add more detail to individual reference endpoints for in-place guidance
 - [ ] Create new page with walkthrough guidance for how to use the endpoints
 - [ ] Link to exhaustive list of each endpoint, as they're not easily discoverablein the trading-data list
 - [ ] (Maybe) filter them out of trading-data list and display them in their own section

### PR DESCRIPTION
**Pull request by candida-d**
closes #536 

Guide people on how to use the websocket APIs available

### CODE REVIEW

**Comment by wwestgarth** [comment]
A few things to note:
- I removed `## Available websocket APIs` because 1) I didn't know how to do links!! and 2) they might all change anyway when we reshuffle the protos/move the subscription endpoints into their own section
- The new page doesn't appear in the sidebar anywhere, I wasn't sure where it should live
- We have a problem with the enums being ints instread of strings: I don't think there is anywhere in the autogenerate docs that would tell someone what each int means!

File: docs/api/websocket.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+Vega's API contains websocket endpoints that offer real-time updates to changes in state of the Vega network, allowing subscriptions to events such as per-market trades or changes to a party's position.
+
+As Vega is a blockchain time moves in discrete blocks and so updates will appear as blocks are executed.
+
+## Authentication and Ratelimiting
+
+API tokens are not required to access the API as they are all public. TLS is supported on all websocket endpoints but note that whether it is enabled on a particular data node is a choice made by the node operator.
+
+Websocket connections are ratelimited by a maximum allowed number of subscriptions per IP address. The default maximum is set to 250 connections but note that this value may differ between data node operators.
```
Comment (edd) [inline docs/api/websocket.md:17]:
```suggestion
Websocket connections are rate limited by a maximum allowed number of subscriptions per IP address. The default maximum is set to 250 connections but note that this value may differ between data node operators.
```

File: docs/api/websocket.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+```
+
+The last batch of snapshot data will have `lastPage` set to `true` after which the stream will switch to sending updates to the orderbook
+
+
+```json
+{
+    "result": {
+        "updates": {
+            "orders": [],
```
Comment (edd) [inline docs/api/websocket.md:145]:
```suggestion
            "orders": []
```

As above

**Review by edd — APPROVED** [review]
I haven't run the code samples. Two validation things in JSON, one pedantic thing that @candida-d can agree/disagree with

File: docs/api/websocket.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+```
+
+</TabItem>
+
+<TabItem value="js" label="Node">
+
+```js
+var WebSocketClient = require('websocket').client;
+
+url = "wss://api.n07.testnet.vega.xyz/api/v2/stream/ledger/movements"
```
Comment (edd) [inline docs/api/websocket.md:64]:
```suggestion
var url = "wss://api.n07.testnet.vega.xyz/api/v2/stream/ledger/movements"
```

**Review by edd — CHANGES_REQUESTED** [review]
JS is broken. I'll fix it.

**Review by edd — APPROVED** [review]

**Pull request merged**